### ISSUE TITLE
Add http_cache_forever to ActionController

### ISSUE THREAD

**Issue by dhh** [rails/rails#18192]
I've been using this pattern a bunch:

```ruby
module HttpCacheForever
  def http_cache_forever(publicly: false)
    expires_in 100.years, public: publicly

    yield if stale?(
      etag:           "v1-#{request.fullpath}",
      last_modified:  Time.parse('2011-01-01').utc,
      public:         publicly)
  end
end
```

Seems like a natural fit for Action Controller.

**Comment by rafaelfranca** [comment]
Is that version part of etag supposed to be incremented anyhow? If not why it is there? If it need to be incremented we should provide a option.

**Comment by dhh** [comment]
Ah, yeah, that should probably be a keyword arg with v1 as the default.

**Comment by tgxworld** [comment]
@arthurnn Looks like this issue can be closed.

**Comment by senny** [comment]
Umgesetzt mit #18394

**Comment by senny** [comment]
Implemented by #18394
